helm/cron_jobs/berkeley-migration-phase-1-cronjob/files/migration-phase-1.sh

```diff
@@ -0,0 +1,171 @@
+import_dump () {
+
+	PREFIX=$1
+	SCHEMA=$2
+
+	echo "Importing ${SCHEMA} archive..."
+
+	echo "Fetching newest dump from ${DUMPS_BUCKET} starting with ${PREFIX}"
+
+	ARCHIVE_DUMP_URI=$(gsutil -o "$KEY_FILE_ARG" ls gs://${DUMPS_BUCKET}/${PREFIX}-*.sql.tar.gz | sort -r | head -n 1);
+	ARCHIVE_DUMP=$(basename $ARCHIVE_DUMP_URI);
+	ARCHIVE_SQL=$(basename $ARCHIVE_DUMP_URI .tar.gz);
+
+	echo "Found lastest dump: " $ARCHIVE_DUMP_URI " . Downloading ...";
+	gsutil -o "$KEY_FILE_ARG" cp $ARCHIVE_DUMP_URI . ;
+
+	echo " Unpacking archive dump";
+	tar -xzvf $ARCHIVE_DUMP;
+	mv $ARCHIVE_SQL ~postgres/;
+	echo "Deleting archive dump";
+	rm -f $ARCHIVE_DUMP;
+	
+	echo "Creating schema and importing archive dump";
+	su postgres -c "cd ~ && echo CREATE DATABASE $SCHEMA | psql";
+	su postgres -c "cd ~ && psql -d $SCHEMA < $ARCHIVE_SQL";
+
+	echo "Deleting archive SQL file";
+	su postgres -c "cd ~ && rm -f $ARCHIVE_SQL";
+	rm -f $ARCHIVE_SQL
+
+}
+
+# Creates target empty schema .
+# Should be used on initial migration steps (when there is no existing partially migrated schema)
+import_dump_frist_time () {
+	SCHEMA=$1
+
+	echo "Creating ${SCHEMA} archive..."
+
+	echo "Fetching newest schema from "
+
+	wget "${CREATE_SCRIPT_URI}"/create_schema.sql
+	wget "${CREATE_SCRIPT_URI}"/zkapp_tables.sql
+	
+	ARCHIVE_SQL=create_schema.sql
+	mv $ARCHIVE_SQL ~postgres/;
+	mv zkapp_tables.sql ~postgres/;
+	echo "Creating schema and importing archive dump";
+	su postgres -c "cd ~ && echo CREATE DATABASE $SCHEMA | psql";
+	su postgres -c "cd ~ && psql -d $SCHEMA < $ARCHIVE_SQL";
+
+	echo "Deleting archive SQL file";
+	su postgres -c "cd ~ && rm -f $ARCHIVE_SQL";
+	rm -f $ARCHIVE_SQL
+
+}
+
+run_first_phase_of_migration() {
+
+
+	echo "Starting migration Phase 1";
+
+	echo "Downloading genesis_ledger";
+	wget $GENESIS_LEDGER_URI -O genesis_ledger.json
+	
+	echo "Running berkeley migration app";
+	
+	if [ -z "${FORK_STATE_HASH}" ]; then
+		mina-berkeley-migration --mainnet-archive-uri ${PG_CONN_STRING}/"${SCHEMA_NAME_FROM}" --migrated-archive-uri ${PG_CONN_STRING}/"${SCHEMA_NAME_TO}" --batch-size 500 --config-file genesis_ledger.json --blocks-bucket "$PRECOMP_BLOCKS_BUCKET" --network "$NETWORK_NAME"  &> "${MIGRATION_LOG}".log
+	else
+		mina-berkeley-migration --mainnet-archive-uri ${PG_CONN_STRING}/"${SCHEMA_NAME_FROM}" --migrated-archive-uri ${PG_CONN_STRING}/"${SCHEMA_NAME_TO}" --batch-size 500 --config-file genesis_ledger.json --blocks-bucket "$PRECOMP_BLOCKS_BUCKET" --network "$NETWORK_NAME" --fork-state-hash $FORK_STATE_HASH &> "${MIGRATION_LOG}".log
+	fi
+	
+	echo "Done running berkeley migration app";
+
+}
+
+service postgresql start;
+su postgres -c "cd ~ && echo ALTER USER postgres WITH PASSWORD \'postgres\' | psql";
+	
+
+install_prereqs
+
+import_dump "${DUMPS_PREFIX_FROM}-archive-dump" $SCHEMA_NAME_FROM
+
+if [[ "$INITIAL_RUN" == "true" ]]
+then 
+	import_dump_frist_time $SCHEMA_NAME_TO
+else 
+	import_dump "${DUMPS_PREFIX_TO}-archive-dump" $SCHEMA_NAME_TO
```

Comment:
Are we taking the output of phase 2 or just the latest output phase 1 here?

Reply:
DUMPS_PREFIX_TO is prefix for migrated database of this is output of phase 2